api/app/Http/Middleware/SlowQueryLoggerMiddleware.php

```diff
@@ -31,15 +31,14 @@ public function handle(Request $request, Closure $next)
         $result = $next($request);
         $end = hrtime(true);
         $elapsedSeconds = ($end - $start) / 1000000000;
+        $requestData = $request->json()->all();
```

Comment:
I think we can get away without having to assign so many variables. Let me know if this doesn't make sense:

```php
$requestData = $request->json()->all();

if ($elapsedSeconds > 5) {
    $logMessage = 'Slow query, '.$elapsedSeconds.', '.($requestData['operationName'] ? $requestData['operationName'] : 'Unnamed operation').', '.request()->headers->get('referer').', '.($requestData['query'] ? json_encode($requestData['query']) : 'No query found');
    $this->logger->info($logMessage);
}
```

Reply:
I actually prefer the readability of what you had originally.  But I would move the variable operations inside the _if_ block so that all the decoding only happens if the time limit is crossed.

Reply:
just changed. putting it back